### PR TITLE
Adds sprites for opening and closing the bolt when using a bolt action rifle (Mosin).

### PR DESCRIPTION
**Pull request by Stokes52**
## What Does This PR Do
* Adds an open bolt sprite for the mosin nagant (bolt action rifle) so that you can visually see whether the bolt is open or closed.
* Updates mosin code so that the sprite will update when the bolt is cycled.
* Small tweak to the mosin description to make it clear that you can attach a knife to it.

## Why It's Good For The Game
Mosin nagants sometimes appear when a wizard summons guns or during admin events.  The mosin is incredibly clunky to use without being able to quickly understand whether the bolt is open or closed. It's very easy to look like an idiot because you try to shoot and the bolt is open.

This simple visual change makes it so that you can easily tell at a glance whether the bolt is closed and ready or still open. It's a small change, but a huge quality of life increase if you ever find yourself in a position where you need to use the mosin.

## Images of changes
New open bolt sprite:
changes

Ingame:
Ingame

## Testing
Load into test server
Spawn mosin nagant
Open and close bolt, to see that sprite updates correctly.
Test firing and loading to see that sprite updates correctly.

## Changelog
:cl:
tweak: Mosin nagant now visually shows whether the bolt is open or closed.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 1]
```diff
@@ -213,7 +213,7 @@
 	inhand_y_dimension = 32
 	mag_type = /obj/item/ammo_box/magazine/internal/boltaction
 	fire_sound = 'sound/weapons/gunshots/gunshot_rifle.ogg'
-	var/bolt_open = 0
+	bolt_open = FALSE
```
Comment (lewcc) [inline code/modules/projectiles/guns/projectile/shotgun.dm:216]: Unless it gets changed somewhere between here and `/obj/item/gun`, this assignment is redundant.

Comment (Stokes52) [reply code/modules/projectiles/guns/projectile/shotgun.dm]: Resolved. Thanks!

**Review by Sirryan2002 — APPROVED** [review]
code looks good

**Review by S34NW — APPROVED** [review]

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -35,6 +35,7 @@
 	var/semicd = 0						//cooldown handler
 	var/weapon_weight = WEAPON_LIGHT
 	var/list/restricted_species
+	var/bolt_open = FALSE
```
Comment (farie82) [inline code/modules/projectiles/gun.dm:38]: I'm not sure it is a good idea to add this variable to the base class. Usually, a good rule of thumb is to not make something generic if it won't be required as such for a while.

My suggestion would be to put this in the mosin definition itself and override `update_icon_state` in accordance with the behaviour you'd expect.

Comment (Stokes52) [reply code/modules/projectiles/gun.dm:38]: Thanks for the review. I will take a look and deconflict this PR as soon as I have time.

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 3]
```diff
@@ -225,8 +226,13 @@
 	else
 		pump_unload(M)
 	bolt_open = !bolt_open
+	icon_state = "moistnugget_open"
+	update_icon()
```
Comment (S34NW) [inline code/modules/projectiles/guns/projectile/shotgun.dm:230]:
```suggestion
	update_icon(UPDATE_ICON_STATE)
```
This is all being handled in `update_icon_state()` now, so you don't need to manually do it here, and you need to tell the code what part of `update_icon()` you want to run.

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 4]
```diff
@@ -213,7 +213,8 @@
 	inhand_y_dimension = 32
 	mag_type = /obj/item/ammo_box/magazine/internal/boltaction
 	fire_sound = 'sound/weapons/gunshots/gunshot_rifle.ogg'
-	var/bolt_open = 0
+	var/bolt_open = FALSE
+	bolt_open = FALSE
```
Comment (S34NW) [inline code/modules/projectiles/guns/projectile/shotgun.dm:217]:
```suggestion
	var/bolt_open = FALSE
```
You don't need two of the same :)

**Review by S34NW — CHANGES_REQUESTED** [review]

**Comment by Stokes52** [comment]
Thank you @S34NW 

@farie82 this PR has been deconflicted.

**Comment by S34NW** [comment]
image

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 5]
```diff
@@ -224,9 +224,13 @@
 		pump_reload(M)
 	else
 		pump_unload(M)
-	bolt_open = !bolt_open
+	icon_state = "moistnugget_open"
```
Comment (S34NW) [inline code/modules/projectiles/guns/projectile/shotgun.dm:227]:
```suggestion
```
this snuck back in, last suggested change before approval

**Review by S34NW — APPROVED** [review]

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 6]
```diff
@@ -224,9 +224,12 @@
 		pump_reload(M)
 	else
 		pump_unload(M)
-	bolt_open = !bolt_open
+	update_icon(UPDATE_ICON_STATE)
 	return 1
```
Comment (hal9000PR) [inline code/modules/projectiles/guns/projectile/shotgun.dm:228]: I am not sure you meant to remove this line? bolt_open never gets set to true

Comment (S34NW) [reply code/modules/projectiles/guns/projectile/shotgun.dm:228]: oh damn

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 7]
```diff
@@ -224,9 +224,12 @@
 		pump_reload(M)
 	else
 		pump_unload(M)
-	bolt_open = !bolt_open
+	update_icon(UPDATE_ICON_STATE)
```
Comment (S34NW) [inline code/modules/projectiles/guns/projectile/shotgun.dm:227]:
```suggestion
	bolt_open = !bolt_open
	update_icon(UPDATE_ICON_STATE)
```

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**